samza-core/src/test/scala/org/apache/samza/system/TestSystemConsumers.scala

```diff
@@ -384,6 +420,30 @@ class TestSystemConsumers {
     def stop {}
     def register { super.register(systemStreamPartition, "0") }
   }
+
+  private class TestStartpointConsumer extends SerializingConsumer with StartpointVisitor {
```

Comment:
Instead of implementing a test object, would it be possible to use a mock and then verify the call to `visit`?